src/Map/Map.h

```diff
@@ -39,12 +43,19 @@ class Map
     // recalculate the owner of a node
     void updateOwner(int index);
 
-public:
     // constructors and destructors
     Map(std::vector<Node> inputNodes, std::string startRegion, std::vector<std::string> regions, std::vector<std::string> continents, std::vector<std::string> players);
     Map(Map const &m);
+
+public:
+
     ~Map();
 
+    // singleton instance
+    static Map* getInstance(std::vector<Node> inputNodes, std::string startRegion, std::vector<std::string> regions, std::vector<std::string> continents, std::vector<std::string> players);
```

Comment:
I'd overload this as `static Map* getInstance()` so we don't need to give all these arguments each time we want to access the map.